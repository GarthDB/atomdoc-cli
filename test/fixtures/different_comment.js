/**
 *  Public: just returns `val`.
 *
 *  * `val` {String} any string.
 *
 *  Returns {String} same `val`.
 */
let list = function (val) {
  return val;
}
/*
 Public: parses a string representing an array into an Array.

 * `val` {String} list string.

 Returns {Array}
 */
list = function (val) {
  const result = val.replace(/(^(?:"|')?\[)|(](?:"|')?$)/g, '').split(',');
  // This is just a regular comment.
  return result.map((item) => item.trim().replace(/(^(?:"|')?)|((?:"|')?$)/g, ''));
};
