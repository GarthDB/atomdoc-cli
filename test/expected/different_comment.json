[
  {
    "originalText": "Public: just returns `val`.\n\n* `val` {String} any string.\n\nReturns {String} same `val`.",
    "visibility": "Public",
    "description": "just returns `val`.",
    "summary": "just returns `val`.",
    "arguments": [
      {
        "name": "val",
        "description": "{String} any string.",
        "type": "String",
        "isOptional": false
      }
    ],
    "returnValues": [
      {
        "type": "String",
        "description": "Returns {String} same `val`."
      }
    ],
    "start": 0,
    "end": 111,
    "className": false,
    "name": "list",
    "definitionLine": 8
  },
  {
    "originalText": "Public: parses a string representing an array into an Array.\n\n* `val` {String} list string.\n\nReturns {Array}",
    "visibility": "Public",
    "description": "parses a string representing an array into an Array.",
    "summary": "parses a string representing an array into an Array.",
    "arguments": [
      {
        "name": "val",
        "description": "{String} list string.",
        "type": "String",
        "isOptional": false
      }
    ],
    "returnValues": [
      {
        "type": "Array",
        "description": "Returns {Array}"
      }
    ],
    "start": 156,
    "end": 274,
    "className": false,
    "name": "list",
    "definitionLine": 18
  }
]
